src/cortex_cli/cortex_cli.py

```diff
@@ -743,14 +751,22 @@ def run(  # pylint: disable=too-many-arguments, too-many-locals, import-outside-
 
         logger.debug('\nInput circuit:\n%s', input_circuit)
 
-        parsed_qubit_mapping = None
-        if qubit_mapping is not None:
-            parsed_qubit_mapping = json.load(qubit_mapping)
+        qubit_mapping = None
+        if qasm_qubit_placement is not None:
+            parsed_qasm_qubit_placement = json.load(qasm_qubit_placement)
+            try:
+                validated_qubit_placement = QasmQubitPlacement(qubit_placement=parsed_qasm_qubit_placement)
+            except ValidationError as ex:
+                raise click.ClickException(f'Invalid qasm_qubit_placement provided: {str(ex)}.')
+            qubit_mapping = {
```

Comment:
instead of building a qubit mapping and then passing it to IQM client let's just map the qubit names in the circuit right away and don't pass any qubit mapping to the client. The circuit object should have a `transform_qubits` method that can do this easily

Reply:
✅